src/main/java/org/mitre/synthea/engine/Logic.java

```diff
@@ -73,18 +76,51 @@ public boolean test(Person person, long time) {
   }
   
   /**
-   * The Date condition type tests the current year being simulated. For example, this may be used
-   * to drive different logic depending on the suggested medications or procedures of different time
-   * periods, or model different frequency of conditions.
+   * The Date condition type tests the current year, month, or date being simulated.
+   * For example, this may be used to drive different logic depending on the suggested
+   * medications or procedures of different time periods, or model different frequency
+   * of conditions.
    */
   public static class Date extends Logic {
+    private String datetype;
     private int year;
+    private int month;
+    private String date; //must be in format yyyy-MM-dd HH:mm:ss.SSS 
     private String operator;
+    private int currentyear;
+    private int currentmonth;
+    private String current;
 
     @Override
     public boolean test(Person person, long time) {
-      int current = Utilities.getYear(time);
-      return Utilities.compare(current, year, operator);
+      if (datetype != null) {
+        switch (datetype) {
+          case "Year":
+            currentyear = Utilities.getYear(time);
+            return Utilities.compare(currentyear, year, operator);
+          case "Month":
+            currentmonth = Utilities.getMonth(time);
+            return Utilities.compare(currentmonth, month, operator);
+          case "Date":
+            DateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
+            java.util.Date testdate = new java.util.Date();
+            java.util.Date currentdate = new java.util.Date();
+            current = Utilities.getDate(time);
```

Comment:
This is kind of a roundabout way to compare two dates, and the more conversions the more likely something can go wrong. You have `time` as a long already, so if you could convert the provided string date into a `long` you could compare those directly, and not have to implement `Utilities.getDate` and `Utilities.compare(Date,Date,String)`.

Reply:
I would need to add `Utilities.compare(long, long, String)` correct?

Reply:
You could, but a `long` can be implicitly widened to a `double`, so if you have 2 `long`s you can use the existing `Utilities.compare(Double, Double, String)` as-is. 
```java
long time = 1;
long time2 = 2;
String operator = ">";
boolean x = Utilities.compare(time, time2, operator);
```

Academic note: there are of course `long`s that cannot be represented as `double`s, but those are all greater than 2^53 ( https://stackoverflow.com/a/1848762 ) which represents a timestamp so far in the future we don't have to worry.